src/providers/address/address.spec.ts

```diff
@@ -57,135 +61,78 @@ describe('AddressProvider', () => {
-      expect(result).toEqual({
-        address: '15qT4RJTjs7GSTReEmgXr4LbMjTVQ51LZA',
-        isValid: true,
-        network: 'livenet',
-        coin: 'btc',
-        translation: {
-          origCoin: 'btc',
-          origAddress: '15qT4RJTjs7GSTReEmgXr4LbMjTVQ51LZA',
-          resultCoin: 'bch',
-          resultAddress: 'CMJLdTeXcv5oLbL4vX1TRZxcyrfuLP7NP2'
-        }
-      });
-    });
-
-    it('should validate if BTC testnet address is correct and return correct values', () => {
-      let BTCTestnetAddress = 'mscoRRUxbicZdUms3EqDr9jwtCmbbPgfcY';
-      let result = addressProvider.validateAddress(BTCTestnetAddress);
-      expect(result).toEqual({
-        address: 'mscoRRUxbicZdUms3EqDr9jwtCmbbPgfcY',
-        isValid: true,
-        network: 'testnet',
-        coin: 'btc',
-        translation: {
-          origCoin: 'btc',
-          origAddress: 'mscoRRUxbicZdUms3EqDr9jwtCmbbPgfcY',
-          resultCoin: 'bch',
-          resultAddress: 'mscoRRUxbicZdUms3EqDr9jwtCmbbPgfcY'
-        }
-      });
-    });
-
-    it('should validate if BCH livenet address is correct and return correct values', () => {
-      let BCHLivenetAddress = 'qq9jk8jskjsmgqnzygwjsghp3knautm2dcnc5e4k7n';
-      let result = addressProvider.validateAddress(BCHLivenetAddress);
-      expect(result).toEqual({
-        address: 'qq9jk8jskjsmgqnzygwjsghp3knautm2dcnc5e4k7n',
-        isValid: true,
-        network: 'livenet',
-        coin: 'bch',
-        translation: {
-          origCoin: 'bch',
-          origAddress: 'qq9jk8jskjsmgqnzygwjsghp3knautm2dcnc5e4k7n',
-          resultCoin: 'btc',
-          resultAddress: '122462mHhZgJqwqGfmmSjFHJcU4pjCKuV9'
-        }
-      });
-    });
-
-    it('should validate if BCH testnet address is correct and return correct values', () => {
-      let BCHTestnetAddress = 'qqr99gt5qdk4qyaxxvzeapgjuxkg6x9ueue95fakj7';
-      let result = addressProvider.validateAddress(BCHTestnetAddress);
-      expect(result).toEqual({
-        address: 'qqr99gt5qdk4qyaxxvzeapgjuxkg6x9ueue95fakj7',
-        isValid: true,
-        network: 'testnet',
-        coin: 'bch',
-        translation: {
-          origCoin: 'bch',
-          origAddress: 'qqr99gt5qdk4qyaxxvzeapgjuxkg6x9ueue95fakj7',
-          resultCoin: 'btc',
-          resultAddress: 'mg6PLV5yyUS6Gy55fJ7f994dQ7RpfJNYC9'
-        }
-      });
-    });
 
-    it('should return invalid values if we send an invalid address', () => {
-      let address = 'invalidAddress';
-      let result = addressProvider.validateAddress(address);
-      expect(result).toEqual({
-        address: 'invalidAddress',
-        isValid: false,
-        network: 'livenet',
-        coin: null,
-        translation: undefined
-      });
+    it('should return null if we send an invalid address', () => {
+      const address = 'invalidAddress';
+      const result = addressProvider.getCoin(address);
+      expect(result).toBeNull();
     });
   });
 
```

Comment:
maybe it would be easy to defined the test vectors:

testVectors:
[
[<string>,<expectedCoin>,<expectedNetwork>],
[<string>,<expectedCoin>,<expectedNetwork>],
...
];

and then iterate

testVectors.forEach( (v)=> {
expect(addressProvider.getNetwork(v[0])).toEqual(v[2]);
expect(addressProvider.getCoin(v[0])).toEqual(v[1]);
});

Reply:
Great idea! Done